### PR TITLE
Emscripten on Windows

### PR DESCRIPTION
**Pull request by Squareys**
Hi @prestomation !

I made some changes to get PhysX 4.1 compile on latest Emscripten on Windows. With newer clang came more warnings that I partially fixed or ignored.

Best,
Jonathan

Note: For the JS wrapper library there are still some duplicate symbol issues that I was not able to resolve. Since I don't need it, I didn't investigate much further.

### CODE REVIEW

File: physx/source/geomutils/src/convex/GuBigConvexData.cpp [hunk 1]
```diff
@@ -149,13 +149,13 @@ bool BigConvexData::Load(PxInputStream& stream)
 	// Load base gaussmap
 //	if(!GaussMap::Load(stream))	return false;
 
-		// Import header
-		if(!ReadHeader('G', 'A', 'U', 'S', Version, Mismatch, stream))
-			return false;
+	// Import header
```
Comment (prestomation) [inline physx/source/geomutils/src/convex/GuBigConvexData.cpp:152]: This file just looks like whitespace changes. Can you revert this? I'd like to limit the changes to existing PhysX code as much to ease ongoing rebasing as well as ease pushing to PhysX once NVidia decides to take this

File: physx/source/physxcooking/src/convex/BigConvexDataBuilder.cpp [hunk 2]
```diff
@@ -83,15 +83,15 @@ bool BigConvexDataBuilder::save(PxOutputStream& stream, bool platformMismatch) c
...
+		return false;
 
-		// Export basic info
+	// Export basic info
 	//	stream.StoreDword(mSubdiv);
-		writeDword(mSVM->mData.mSubdiv, platformMismatch, stream);		// PT: could now write Word here
+	writeDword(mSVM->mData.mSubdiv, platformMismatch, stream);		// PT: could now write Word here
 	//	stream.StoreDword(mNbSamples);
-		writeDword(mSVM->mData.mNbSamples, platformMismatch, stream);	// PT: could now write Word here
+	writeDword(mSVM->mData.mNbSamples, platformMismatch, stream);	// PT: could now write Word here
```
Comment (prestomation) [inline physx/source/physxcooking/src/convex/BigConvexDataBuilder.cpp:94]: Same with this, this all looks like whitespace changes

File: physx/source/physxextensions/src/serialization/Xml/SnXmlMemoryPool.h [hunk 3]
```diff
@@ -323,7 +323,7 @@ namespace physx {
 				m8ItemPool.deallocate(inMemory);
 			else
 			*/
-				mVariablePool.deallocate(inMemory);
+			mVariablePool.deallocate(inMemory);
```
Comment (prestomation) [inline physx/source/physxextensions/src/serialization/Xml/SnXmlMemoryPool.h:326]: same here

File: physx/source/compiler/cmake/emscripten/PhysXWebBindings.cmake [hunk 4]
```diff
@@ -103,3 +107,5 @@ TARGET_INCLUDE_DIRECTORIES(PhysXWebBindings
 
 # name the output library 'physx' as this is really a union of all of js-bound PhysX
 set_target_properties(PhysXWebBindings PROPERTIES OUTPUT_NAME "physx.${CMAKE_BUILD_TYPE}")
+
+SET_TARGET_PROPERTIES(PhysXWebBindings PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
```
Comment (prestomation) [inline physx/source/compiler/cmake/emscripten/PhysXWebBindings.cmake:111]: Interesting, did you need this?

File: physx/source/geomutils/src/gjk/GuGJKType.h [hunk 5]
```diff
@@ -151,7 +151,7 @@ namespace Gu
 
 		virtual Ps::aos::Vec3V getCenter() const { return mAToB.transform(getConvex<Convex>().getCenter()); }
 
-		PX_FORCE_INLINE Ps::aos::PsMatTransformV& getRelativeTransform(){ return mAToB; }
+		PX_FORCE_INLINE const Ps::aos::PsMatTransformV& getRelativeTransform() const { return mAToB; }
```
Comment (prestomation) [inline physx/source/geomutils/src/gjk/GuGJKType.h:154]: Is this the only clang-related change?

**Review by prestomation — CHANGES_REQUESTED** [review]
Thanks alot! mostly just some questions here to ensure I understand

File: physx/source/geomutils/src/convex/GuBigConvexData.cpp (hunk 1 above)
Comment (Squareys) [reply physx/source/geomutils/src/convex/GuBigConvexData.cpp:152]: As mentioned in the commit messages, these are "misleading indentation" warnings in clang, that are treaded as errors. I can instead ignore the warning, if you'd prefer that.

File: physx/source/geomutils/src/gjk/GuGJKType.h (hunk 5 above)
Comment (Squareys) [reply physx/source/geomutils/src/gjk/GuGJKType.h:154]: Yes, this and the misleading indentation warning fixes were the only changes to the code that were necessary. Everything else was ignoring warnings in CMake or linking issues.

File: physx/source/compiler/cmake/emscripten/PhysXWebBindings.cmake (hunk 4 above)
Comment (Squareys) [reply physx/source/compiler/cmake/emscripten/PhysXWebBindings.cmake:111]: Yes: https://github.com/prestomation/PhysX/pull/6/commits/66cbfc6f6de65aa58ad2012d3314a88995d0e19e

File: physx/source/geomutils/src/convex/GuBigConvexData.cpp (hunk 1 above)
Comment (Squareys) [reply physx/source/geomutils/src/convex/GuBigConvexData.cpp:152]: See https://github.com/prestomation/PhysX/pull/6/commits/c656a9e9a5f4572809311e49a5494625ba72bcf4

**Comment by Squareys** [comment]
@prestomation Awesome, thanks for merging! Hope NV will eventuell merge this upstream :)

**Pull request merged**